statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/httpfn/HttpFunction.java

```diff
@@ -109,9 +137,17 @@ private void onAsyncResult(
     handleInvocationResponse(context, invocationResult);
     InvocationBatchRequest.Builder nextBatch = getNextBatch();
     if (nextBatch == null) {
-      hasInFlightRpc.clear();
+      // the async request was completed, and there is nothing else in the batch
+      // so we clear the requestState.
+      requestState.clear();
       return;
     }
+    // an async request was just completed, but while it was in flight we have
+    // accumulated a batch, we now proceed with:
+    // a) clearing the batch from our own persisted state (the batch moves to the async operation
+    // state)
+    // b) sending the accumulated batch to the remote function.
+    requestState.set(0);
```

Comment:
Should this actually be `requestState.set(nextBatch.size())`?
Or rather, we don't set it to 0 because at this point request state is already == accumulated batch size.

Reasoning is:
In `onRequest` method, as I understood it from the naming, the `inflightOrBatched` (counter obtained from `requestState`) reflects total number of "pending" records, regardless of whether they are buffered or in-flight.
Setting `requestState` to 0 here breaks that semantic.

Reply:
If that's the case, then probably `maxBatchSize` should be renamed to `maxPendingSize`.

Reply:
On the other hand, if the intended semantics is that we only backpressure when local accumulated batch exceeds threshold, and not accounting number of in-flights (records, not requests; a batch request would have multiple records), then probably `inflightOrBatched` should be renamed to `batchedSize`.

Reply:
So just to sum up 😄 
I'm trying to clarify the intended backpressuring semantics w.r.t. some of the variable namings here.
There seems to be some inconsistency in the naming?